lib/tx_build/claim_predicate.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Stellar.TxBuild.ClaimPredicate do
+  @moduledoc """
+  `ClaimPredicate` struct definition.
+  """
+  alias Stellar.TxBuild.{ClaimPredicates, OptionalClaimPredicate}
+  alias StellarBase.XDR.{ClaimPredicate, ClaimPredicateType, Int64, Void}
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type validation :: {:ok, any()} | {:error, any()}
+  @type predicate :: :unconditional | :conditional
+  @type type :: :time | :and | :or | :not
+  @type time_type :: :relative | :absolute
+  @type value :: t() | list(t()) | pos_integer()
```

Comment:
@einerzg IMO, you should use the `Stellar.TxBuild.ClaimPredicates` type instead of `list(t())`.